common4j/src/main/com/microsoft/identity/common/java/cache/AbstractAccountCredentialCache.java

```diff
@@ -276,6 +276,35 @@ protected List<Credential> getCredentialsFilteredByInternal(@NonNull final List<
         return matchingCredentials;
     }
 
+    /**
+     * The application identifier field for access token records previously included a SHA-1 signing certificate hash.
+     * This method clears such tokens, as the application identifier should now contain a SHA-512 signing certificate hash.
+     */
+    public void clearSha1ApplicationIdentifierAccessTokens() {
+        final String methodTag = TAG + ":clearSha1ApplicationIdentifierAccessTokens";
+        for (final Credential credential : getCredentials()) {
```

Comment:
how much perf does this cost (given getCredentials is an I/O operation)

OneAuth is pretty sensitive about it and i'm wondering if we should add a flag (to the storage) to mark if the "clean sha1" operation is already done.

Reply:
Gave your suggested implementation a go. let me know what you think